src/js/popup/popup.js

```diff
@@ -941,82 +941,87 @@
         },
       },
       webcompat: {
-        init: () => {
-          popup.panel.webcompat.setURLwithIssue();
-          popup.panel.webcompat.setRequiredCheckboxListeners();
-          popup.panel.webcompat.showReportInputOtherTextField();
+        handleReportIssueFormSubmission: async (event, formData) => {          
+          event.preventDefault();
 
-          const reportForm = document.querySelector(".report-issue-content");
-          reportForm.addEventListener("submit", async (event) => {
-            await popup.panel.webcompat.handleReportIssueFormSubmission(event);
+          // Do not submit if the form is not valid
+          if (!formData.form.dataset.formIsValid) {
+            return false;
+          }
+          
+          const data = new FormData(event.target);
+          const reportData = Object.fromEntries(data.entries());
+          reportData.user_agent = await getBrowser();
+
+          Object.keys(reportData).forEach(function (value) {
+            // Switch "on" to true
+            if (reportData[value] === "on") {
+              reportData[value] = true;
+            }
+            // Remove from report if empty string
+            if (reportData[value] === "") {
+              delete reportData[value];
+            }
           });
 
-          const reportContinueButton =
-            document.querySelector(".report-continue");
-          reportContinueButton.addEventListener(
-            "click",
-            popup.events.backClick,
-            false
-          );
+          // Clean URL data to add "http://" before it if the custom input doesn't contain a HTTP protocol
+          if (
+            !(
+              reportData.issue_on_domain.startsWith("http://") ||
+              reportData.issue_on_domain.startsWith("https://")
+            )
+          ) {
+            reportData.issue_on_domain = "http://" + reportData.issue_on_domain;
+          }
+
+          await browser.runtime.sendMessage({
+            method: "postReportWebcompatIssue",
+            description: reportData,
+          });
         },
-        toggleRequiredCheckboxListeners: (toggleOverride = null) => {
-          const checkboxes = document.querySelectorAll('.report-section ul li input');
+        init: () => {
+          const formData = {
+            form: document.querySelector(".report-issue-content"),
+            reportIssueSubmitBtn: document.querySelector(".report-issue-submit-btn"),
+            inputFieldUrl: document.querySelector('input[name="issue_on_domain"]'),
+            inputFieldOtherDetails: document.querySelector('input[name="other_issue"]'),
+            checkboxes: document.querySelectorAll('.report-section ul li input'),
+            otherTextField: document.querySelector('input[name="other_issue"]'),
+            otherCheckbox: document.querySelector('input[name="issue-case-other"]'),
+          }
```

Comment:
Moved all DOM setting into a single object, and pass it around to the sub functions.